memory/src/main/scala/filodb.memory/MemFactory.scala

```diff
@@ -311,6 +311,21 @@ class BlockMemFactory(blockStore: BlockManager,
     metaAddr
   }
 
+  /**
+    * Discards the current metadata span, to be called when a failure occurs during allocation.
+    * The blocks are then marked as reclaimable, and they are fully reclaimed later by the
+    * headroom task.
+    */
+  def discardMetaSpan(): Unit = {
+    if (metadataSpanActive) {
+      metadataSpan.foreach { blk =>
+        blk.markReclaimable()
```

Comment:
Is this correct ? Should it be marked as reclaimable ?
`markFullBlocksAsReclaimable` needs to be checked?

Reply:
I had that check originally, but it seemed unnecessary.

Reply:
When discard is called, it's due to an exception. Keeping blocks around that might have nothing in it doesn't make sense. In practice, it won't make a difference because the mark full mode is only used with ODP, and discard isn't needed for ingestion.

Reply:
I'll flip it back, at least for consistency.